src/esperanza/finalizationstate.cpp

```diff
@@ -160,16 +162,15 @@ void FinalizationState::InstaJustify() {
 
 void FinalizationState::IncrementDynasty() {
   // finalized epoch is m_current_epoch - 2 because:
-  // finalized (0) - justified (1) - votes to justify (2)
+  // justified(0) - finalized(1) - votes to justify(2) - m_current_epoch(3)
```

Comment:
>Both finalized actually.
technically everything is finalized up into the last finalized point but we keep the mark "finalized/justified/nothing" at the time that epoch finished. That's why `justified(0)` is only justified and doesn't have a finalized flag.

I left a comment as relative to what is the number `m_current_epoch` holds but if you found it confusing, I'll drop `justified(0)`

Reply:
> justified(0) - finalized(1)

Both finalized actually.

> votes to justify(2) - m_current_epoch(3)

The whole function is too context dependent, and must be aligned with `InitializeEpoch` in only one way. I'd explain this in comment, otherwise it's not obvious why `votes to justify(2)` and `m_current_epoch(3)` couldn't be the same epoch.

Also, I'd change a comment to not mention epoch before finalization, like:

```
// finalized epoch is m_current_epoch - 2 because:
// finalized(0) - votes to justify(1) - m_current_epoch(2)
```

Otherwise it wasn't clear how "m_current_epoch - **2**" explained over "m_current_epoch(**3**)".

Reply:
Corrected the comment